File: tests/sample.rb
# Sample Ruby Code: Simple CLI To-Do App

def display_tasks(tasks)
    puts "\nTo-Do List:"
    if tasks.empty?
      puts "  No tasks yet!"
    else
      tasks.each_with_index do |task, index|
        status = task[:completed] ? "[X]" : "[ ]"
        puts "  #{index + 1}. #{status} #{task[:text]}"
      end
    end
  end
  
  def add_task(tasks)
    print "Enter a task: "
    task_text = gets.chomp.strip
    if task_text.empty?
      puts "Task cannot be empty."
    else
      tasks << { text: task_text, completed: false }
      puts "Task added!"
    end
  end
  
  def delete_task(tasks)
    print "Enter task number to delete: "
    task_num = gets.chomp.to_i
    if task_num.between?(1, tasks.size)
      tasks.delete_at(task_num - 1)
      puts "Task deleted!"
    else
      puts "Invalid task number."
    end
  end
  
  def toggle_task_completion(tasks)
    print "Enter task number to toggle completion: "
    task_num = gets.chomp.to_i
    if task_num.between?(1, tasks.size)
      tasks[task_num - 1][:completed] = !tasks[task_num - 1][:completed]
      puts "Task status updated!"
    else
      puts "Invalid task number."
    end
  end
  
  def main
    tasks = []
    loop do
      puts "\nOptions:"
      puts "  1. View tasks"
      puts "  2. Add task"
      puts "  3. Delete task"
      puts "  4. Toggle task completion"
      puts "  5. Exit"
  
      print "Choose an option: "
      choice = gets.chomp
  
      case choice
      when "1"
        display_tasks(tasks)
      when "2"
        add_task(tasks)
      when "3"
        delete_task(tasks)
      when "4"
        toggle_task_completion(tasks)
      when "5"
        puts "Goodbye!"
        break
      else
        puts "Invalid choice. Please try again."
      end
    end
  end
  
  main
  

File: tests/sample.py
# Sample Python Code: Simple CLI To-Do App

def display_tasks(tasks):
    print("\nTo-Do List:")
    if not tasks:
        print("  No tasks yet!")
    for i, task in enumerate(tasks, 1):
        status = "[X]" if task['completed'] else "[ ]"
        print(f"  {i}. {status} {task['text']}")

def add_task(tasks):
    task_text = input("Enter a task: ").strip()
    if task_text:
        tasks.append({"text": task_text, "completed": False})
        print("Task added!")
    else:
        print("Task cannot be empty.")

def delete_task(tasks):
    try:
        task_num = int(input("Enter task number to delete: "))
        if 1 <= task_num <= len(tasks):
            tasks.pop(task_num - 1)
            print("Task deleted!")
        else:
            print("Invalid task number.")
    except ValueError:
        print("Please enter a valid number.")

def toggle_task_completion(tasks):
    try:
        task_num = int(input("Enter task number to toggle completion: "))
        if 1 <= task_num <= len(tasks):
            tasks[task_num - 1]['completed'] = not tasks[task_num - 1]['completed']
            print("Task status updated!")
        else:
            print("Invalid task number.")
    except ValueError:
        print("Please enter a valid number.")

def main():
    tasks = []
    while True:
        print("\nOptions:")
        print("  1. View tasks")
        print("  2. Add task")
        print("  3. Delete task")
        print("  4. Toggle task completion")
        print("  5. Exit")
        
        choice = input("Choose an option: ")
        if choice == "1":
            display_tasks(tasks)
        elif choice == "2":
            add_task(tasks)
        elif choice == "3":
            delete_task(tasks)
        elif choice == "4":
            toggle_task_completion(tasks)
        elif choice == "5":
            print("Goodbye!")
            break
        else:
            print("Invalid choice. Please try again.")

if __name__ == "__main__":
    main()
